optimizations.md

```diff
@@ -0,0 +1,288 @@
+KEVM Optimizations
+==================
+
+These optimizations work on the LLVM and Haskell backend and are generated by the script `./optimizer/optimizations.sh`.
+
+```k
+requires "evm.md"
+
+module EVM-OPTIMIZATIONS-LEMMAS [kore, symbolic]
+    imports EVM
+
+    rule #sizeWordStack(WS           , N) => #sizeWordStack(WS, 0) +Int N requires N =/=Int 0                [simplification]
+    rule #sizeWordStack(WS [ I := _ ], N) => #sizeWordStack(WS, N)        requires I <Int #sizeWordStack(WS) [simplification]
+
+    rule #stackUnderflow(WS, N) => false requires N <=Int #sizeWordStack(WS) [simplification]
```

Comment:
@daejunpark in particular, this is the trust base for both generating the optimizations, and proving them correct w.r.t. the semantics.

Note that we _do not_ trust any of the other lemmas is `tests/specs/**` for generating or proving these optimizations.